Observe some -Xlint recommendations

som-snytt opened the pull request:
Not all unused things are removed outright.

lrytz commented on src/reflect/scala/reflect/internal/StdNames.scala:
```diff
@@ -351,7 +351,7 @@ trait StdNames {
     val REIFY_FREE_PREFIX: NameType        = "free$"
     val REIFY_FREE_THIS_SUFFIX: NameType   = "$this"
-    val REIFY_FREE_VALUE_SUFFIX: NameType  = "$value"
+    val REIFY_FREE_VALUE_SUFFIX: NameType  = "$" + "value"
```
not sure about those.. people will wonder why the definitions here are not all the same style.

som-snytt replied:
I'll check if intrinsic `s` interpolator does a constant.

s/instrinsic/instringsic/

lrytz commented on src/compiler/scala/reflect/reify/phases/Reshape.scala:
```diff
@@ -235,6 +235,7 @@ trait Reshape {
     }
 
+    /*
     private def toPreTyperLazyVal(ddef: DefDef): ValDef = {
```
let's remove this, and unused code in general in this PR.

som-snytt replied:
OK. I'm a hoarder.

lrytz commented on src/compiler/scala/tools/cmd/Interpolation.scala:
```diff
@@ -33,15 +33,15 @@ trait Interpolation {
   /** A simple template for generating bash completion functions.
    */
-  lazy val bashTemplate = """
+  lazy val bashTemplate = s"""
```
i'm not a big fan of this, but OK if you insist :)

som-snytt replied:
This was the one I liked! As in, this is canonical.

som-snytt replied:
Confirmed that intrinsic s just yields a constant string, as desired.

lrytz commented on src/library/scala/Predef.scala:
```diff
@@ -616,7 +616,7 @@ object Predef extends LowPriorityImplicits {
   object <:< {
     /** If `A <: B` and `B <: A`, then `A = B` (subtyping is antisymmetric) */
-    def antisymm[A, B](implicit l: A <:< B, r: B <:< A): A =:= B = =:=.singleton.asInstanceOf[A =:= B]
+    def antisymm[A, B](implicit @deprecated("unused","shim") l: A <:< B, @deprecated("unused","shim") r: B <:< A): A =:= B = =:=.singleton.asInstanceOf[A =:= B]
```
I'd avoid doing this, I find looking at this much more annoying than the lint warning (which we never see). Especially in the standard library, but even in the compiler.

som-snytt replied:
Yes, I'll rebase over the PR which doesn't warn. Because I see the warnings. At night in my dreams.

som-snytt replied:
Actually, this was at the beginning of the edits, when I thought I wanted to minimize the compiler output, but by the end it wasn't so bad, with deprecations turned off of course.

lrytz commented on src/library/scala/collection/View.scala:
```diff
@@ -143,7 +143,7 @@ object View extends IterableFactory[View] {
 
   /** A view that partitions an underlying collection into two views */
-  @SerialVersionUID(3L)
+  //@SerialVersionUID(3L)
```
I think the bug is the other way here, so we could leave this for now. https://github.com/scala/bug/issues/10945

som-snytt replied:
As I suspected, thx.

lrytz commented on src/library/scala/collection/immutable/ChampHashMap.scala:
```diff
@@ -182,6 +183,7 @@ private class BitmapIndexedMapNode[K, +V](val dataMap: Int, val nodeMap: Int, va
 
     predicate1 && predicate2 && predicate3
   }
+  */
```
👍, this makes sense to keep in comments for a while.

lrytz reviewed (changes requested):
I think we can remove unused code in general, the only exception I spotted is the check methods in ChampHashSet/Map, these probably make sense to keep in comments.

som-snytt commented on src/compiler/scala/tools/nsc/io/Socket.scala:
```diff
@@ -37,7 +37,7 @@ class Socket(jsocket: JSocket) extends Streamable.Bytes with Closeable {
 
   def printWriter()                         = new PrintWriter(outputStream(), true)
-  def bufferedReader(implicit codec: Codec) = new BufferedReader(new InputStreamReader(inputStream()))
+  def bufferedReader(implicit codec: Codec) = new BufferedReader(new InputStreamReader(inputStream(), codec.decoder))
```
A potential bug!

som-snytt commented on build.sbt:
```diff
@@ -158,6 +158,8 @@ lazy val commonSettings = instanceSettings ++ clearSourceAndResourceDirectories
   },
   fork in run := true,
+  //scalacOptions += "-Xlint:-nullary-override,-by-name-right-associative,-inaccessible,_",
+  //scalacOptions ++= Seq("-Xmaxerrs", "5", "-Xmaxwarns", "5"),
```
placeholders.

som-snytt commented on src/compiler/scala/reflect/reify/phases/Reshape.scala:
```diff
@@ -248,6 +249,7 @@ trait Reshape {
       val mods2 = toPreTyperModifiers(mods1, ddef.symbol)
       ValDef(mods2, name1, tpt0, extractRhs(rhs0))
     }
+    */
```
First, do no harm.

som-snytt commented on src/compiler/scala/tools/nsc/symtab/SymbolTrackers.scala:
```diff
@@ -182,9 +182,7 @@ trait SymbolTrackers {
-        val treeStrings = symMap(sym) map { t =>
-          "%10s: %s".format(t.shortClass, t)
-        }
+        val treeStrings = symMap(sym).map(t => s"${t.shortClass}%10s: $t")
```
For a moment, I forgot the s interpolator. Luckily, -Xlint told me.

Actually, -Xlint complained about `:+` on L182. Still need to follow up.

som-snytt commented on src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala:
```diff
@@ -1209,7 +1208,7 @@ abstract class ClassfileParser {
       in.bp = runtimeAnnotStart
       val numAnnots = u2
-      var i = 0
+      val i = 0
```
Not obvious whether it's a cut/paste bug.

lrytz replied:
This is an oversight, `i` should be incremented in the loop. https://github.com/scala/scala/pull/6816

som-snytt replied:
Reverted to var.

som-snytt replied:
You said you reverted it. Maybe you inverted it.

som-snytt commented on src/compiler/scala/tools/nsc/transform/patmat/PatternMatching.scala:
```diff
@@ -65,7 +65,7 @@ trait PatternMatching extends Transform
           // but for the rest of us pass in top as the expected type to avoid waste.
           val pt = if (origTp <:< definitions.AnyTpe) definitions.AnyTpe else WildcardType
-          localTyper.typed(translated, definitions.AnyTpe) setType origTp
+          localTyper.typed(translated, pt) setType origTp
```
`-Xlint` would have told you that. -- Paul

som-snytt commented on src/compiler/scala/tools/nsc/typechecker/PatternTypers.scala:
```diff
@@ -44,7 +44,7 @@ trait PatternTypers {
     import infer._
 
-    private def unit = context.unit
+    //private def unit = context.unit
```
can probably zap aliases.

som-snytt commented on src/compiler/scala/tools/nsc/typechecker/Typers.scala:
```diff
@@ -5116,7 +5115,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       }
 
-      def typedTypeSelectionQualifier(tree: Tree, pt: Type = AnyRefTpe) =
+      def typedTypeSelectionQualifier(tree: Tree, pt: Type /*= AnyRefTpe*/) =
```
documentary default, removing unused default.

lrytz replied:
Happy to have the default removed

som-snytt replied:
Made non-documentary.

som-snytt commented on src/compiler/scala/tools/nsc/typechecker/Typers.scala:
```diff
@@ -5557,7 +5556,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
 
       def typedSingletonTypeTree(tree: SingletonTypeTree) = {
-        val refTyped = typedTypeSelectionQualifier(tree.ref, WildcardType )
+        val refTyped = typedTypeSelectionQualifier(tree.ref, WildcardType)
```
collateral whitespace change!

som-snytt commented on src/library/scala/Predef.scala:
```diff
@@ -616,7 +616,7 @@ object Predef extends LowPriorityImplicits {
   object <:< {
     /** If `A <: B` and `B <: A`, then `A = B` (subtyping is antisymmetric) */
-    def antisymm[A, B](implicit l: A <:< B, r: B <:< A): A =:= B = =:=.singleton.asInstanceOf[A =:= B]
+    def antisymm[A, B](implicit @deprecated("unused","shim") l: A <:< B, @deprecated("unused","shim") r: B <:< A): A =:= B = =:=.singleton.asInstanceOf[A =:= B]
```
Made a PR to ignore these cases.

som-snytt commented on src/library/scala/collection/ArrayOps.scala:
```diff
@@ -1129,7 +1129,7 @@ final class ArrayOps[A](val xs: Array[A]) extends AnyVal {
       def mkRowBuilder() = ArrayBuilder.make[B](ClassTag[B](aClass.getComponentType))
       val bs = new ArrayOps(asArray(xs(0))).map((x: B) => mkRowBuilder())
-      var j = 0
+      //var j = 0
```
safe to del.

som-snytt commented on src/library/scala/collection/View.scala:
```diff
@@ -143,7 +143,7 @@ object View extends IterableFactory[View] {
 
   /** A view that partitions an underlying collection into two views */
-  @SerialVersionUID(3L)
+  //@SerialVersionUID(3L)
```
These warnings in the build should be resolved. Also, maybe the warning should sit behind a flag. `-Xlint:serial`? `-Ywarn-serial`? `-Ywarn-srsly`?

som-snytt commented on src/library/scala/concurrent/Channel.scala:
```diff
@@ -17,11 +17,11 @@ package scala.concurrent
  */
 class Channel[A] {
-  class LinkedList[A] {
+  private class LinkedList {
```
we're all friends here.

som-snytt commented on src/reflect/scala/reflect/api/Trees.scala:
```diff
@@ -2165,7 +2165,8 @@ trait Trees { self: Universe =>
    */
-  @deprecated("use q\"new ${sym.toType}(..$args)\" instead", "2.10.1")
+  private[this] final val depstr_New = "use q\"new ${sym." + "toType}(..$args)\" instead"
+  @deprecated(depstr_New, "2.10.1")
```
implicitNotFound gets a pass, but actually all ordinary annotations? I think macros supplying constants would be OK.

som-snytt commented on src/reflect/scala/reflect/api/TypeTags.scala:
```diff
@@ -300,6 +300,7 @@ trait TypeTags { self: Universe =>
   /* @group TypeTags */
   // This class only exists to silence MIMA complaining about a binary incompatibility.
   // Only the top-level class (api.PredefTypeCreator) should be used.
+  @deprecated("This class only exists to silence MIMA complaining about a binary incompatibility.", since="forever")
```
didn't work

som-snytt commented on src/reflect/scala/reflect/internal/Types.scala:
```diff
@@ -2501,6 +2501,7 @@ trait Types
         super.prefixString
     )
     // Suppressing case class copy method which risks subverting our single point of creation.
+    @deprecated("Suppressing case class copy method", since="forever")
```
also didn't work.

som-snytt commented on src/reflect/scala/reflect/internal/transform/UnCurry.scala:
```diff
@@ -108,7 +108,7 @@ trait UnCurry {
     // so we still see repeated parameter types (uncurry replaces them with Seq)
     val isRepeated = origSym.info.paramss.flatten.map(sym => definitions.isRepeatedParamType(sym.tpe))
-    val oldPs = newInfo.paramss.head
+    val _ = newInfo.paramss.head
```
wary of side effects, left it in.

som-snytt commented on src/reflect/scala/reflect/io/IOStats.scala:
```diff
@@ -1,7 +1,7 @@
 package reflect.io
 
-import scala.reflect.internal.util.Statistics
+//import scala.reflect.internal.util.Statistics
```
deletable

som-snytt replied:
actually all that file is commented out.

som-snytt commented on src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala:
```diff
@@ -944,10 +944,10 @@ class ILoop(config: ShellConfig, inOverride: BufferedReader = null,
     * Run with output muted both from ILoop and from the intp reporter.
     */
-  private def interpretPreamble = {
+  private def interpretPreamble() = {
```
left most "parens mean side effects" alone.

som-snytt commented:
Feedback is addressed. Hopefully will be merged before it fails to merge.

som-snytt commented on src/reflect/scala/reflect/internal/StdNames.scala:
```diff
@@ -351,7 +351,7 @@ trait StdNames {
     val REIFY_FREE_PREFIX: NameType        = "free$"
     val REIFY_FREE_THIS_SUFFIX: NameType   = "$this"
-    val REIFY_FREE_VALUE_SUFFIX: NameType  = "$value"
+    val REIFY_FREE_VALUE_SUFFIX: NameType  = s"$$value"      // looks like missing interpolator due to `value` in scopre
```
Unfortunately, `StringContext` name hides the interpolator. Ironic!

som-snytt committed:
Avoid naming a term StringContext if it isn't one

The consequence of introducing `StringContext` is
that you can't use it as one. Just rename it to
something that says it's just a name.

som-snytt committed:
Avoid appearance of interpolation in StdNames

A few `$name` happen to look like an interpolation of
a value `name`. Avoid a lint warning by using an
interpolator. To avoid the idiom for dollar literal,
let the interpolator also add the dollar. Another
day, another dollar. Also, name it the n-terpolator.

som-snytt committed:
Observe some -Xlint recommendations

Not all unused things are removed outright.

som-snytt committed:
Add intended but missing Serializable in View

lrytz commented on src/reflect/scala/reflect/internal/StdNames.scala:
```diff
@@ -351,7 +358,7 @@ trait StdNames {
     val MIRROR_UNTYPED: NameType           = "$m$untyped"
     val REIFY_FREE_PREFIX: NameType        = "free$"
     val REIFY_FREE_THIS_SUFFIX: NameType   = "$this"
```
was there no warning here?

som-snytt replied:
Sharp eyes! I'll wonder if this isPrivateLocal when I'm fully awake.

som-snytt replied:
I need a couple more days of sleep after the national election, then I'll wonder this. Or `"$this"`.

lrytz commented on src/reflect/scala/reflect/internal/StdNames.scala:
```diff
@@ -312,6 +314,11 @@ trait StdNames {
+    private class NameContext(s: String) {
+      def n(args: Any*): TermName = { require(args.isEmpty) ; createNameType("$" + s) }
+    }
+    private def StringContext(parts: String*) = { require(parts.size == 1) ; new NameContext(parts.head) }
```
👍 haven't seen this before

som-snytt replied:
The puzzler guy didn't like my one-liner StringContext puzzler. :(  It was too improbable.

som-snytt replied:
For some reason, we aren't allowed to have fun things.

lrytz reviewed (commented):
> Hopefully will be merged before it fails to merge.

😢

lrytz commented:
LGTM otherwise, let's merge quickly after the rebase.
